static/app/types/system.tsx

```diff
@@ -234,29 +234,195 @@ export type Broadcast = {
   title: string;
 };
 
-export type SentryServiceIncident = {
-  affectedComponents: Array<{
-    name: string;
-    status: 'degraded_performance' | 'partial_outage' | 'major_outage' | 'operational';
-    updatedAt: string;
-  }>;
-  createdAt: string;
-  id: string;
+// XXX(epurkhiser): The components list can be generated using jq
+//
+// curl -s https://status.sentry.io/api/v2/components.json \
+// | jq -r '
+//   .components
+//   | map({key: (.name | gsub( "[^a-zA-Z]"; "_") | ascii_upcase ), value:.id})
+//   | map("\(.key) = \"\(.value)\",")
+//   | .[]'
+// | sort
+
+/**
+ * Mapping of components to IDs
+ *
+ * Should be kept in sync with https://status.sentry.io/api/v2/components.json
+ */
+export enum StatusPageComponent {
```

Comment:
can we use const enum to compile the unused ones away

```suggestion
export const enum StatusPageComponent {
```